### ISSUE TITLE
Vercel serves an XML file unstead of the home page of the blog

### ISSUE THREAD

**Issue by cosydney** [timlrx/tailwind-nextjs-starter-blog#251]
**Describe the bug**
On most (5 times out of 6) production builds on vercel, an XML file is being deployed as the home page of the blog. 

**To Reproduce**
I don't have any way to reproduce it as of now.

**Expected behavior**
I expect the home page to render as an HTML with CSS..

**Screenshots**
<img width="1140" alt="CleanShot 2021-09-28 at 11 03 32@2x" src="https://user-images.githubusercontent.com/10532381/135057838-3b6005b3-416e-4797-8c5d-43a14e76be7e.png">

**System Info (if dev / build issue):**
 - OS: [e.g. iOS] => Vercel default
 - Node version (please ensure you are using 12+) => Node 14.x
 - Npm version => Yarn

**Additional context**
You can see for yourself on https://blog.axolo.co/blog. If you go on https://blog.axolo.co/blog/p and then click on the logo on the top left, it will take you to https://blog.axolo.co/blog without troubles.

here is my next.config.js file: 

I wonder if it comes from webpack5. 

```javascript
const withBundleAnalyzer = require('@next/bundle-analyzer')({
  enabled: process.env.ANALYZE === 'true',
})

module.exports = withBundleAnalyzer({
  basePath: '/blog',
  reactStrictMode: true,
  pageExtensions: ['js', 'jsx', 'md', 'mdx'],
  future: {
    webpack5: true,
  },
  webpack: (config, { dev, isServer }) => {
    config.module.rules.push({
      test: /\.(png|jpe?g|gif|mp4)$/i,
      use: [
        {
          loader: 'file-loader',
          options: {
            publicPath: '/_next',
            name: 'static/media/[name].[hash].[ext]',
          },
        },
      ],
    })

    config.module.rules.push({
      test: /\.svg$/,
      use: ['@svgr/webpack'],
    })

    if (!dev && !isServer) {
      // Replace React with Preact only in client production build
      Object.assign(config.resolve.alias, {
        react: 'preact/compat',
        'react-dom/test-utils': 'preact/test-utils',
        'react-dom': 'preact/compat',
      })
    }

    return config
  },
})

```

It might be an easy fix, was wondering if anyone had a clue of where I should look at ! thank you for your time 🙏

**Comment by timlrx** [comment]
Sounds like you are using an older version of the template that has this issue. See related issue - https://github.com/timlrx/tailwind-nextjs-starter-blog/issues/16. If so, there are two ways of fixing it:

1)  Change https://github.com/timlrx/tailwind-nextjs-starter-blog/blob/master/scripts/generate-sitemap.js#L50 to `sitemap.xml` or something other than `index.xml`. You can then add a 301 redirect to that page if you were using `index.xml` previously. This is the current approach the template takes.

2) Patch to manually map `/` to `index/` - https://github.com/timlrx/timlrx.com/pull/17. This has some issues while developing, you might need to refresh, but otherwise, it works.

**Comment by cosydney** [comment]
Appreciate the quick response, let me try that !

**Comment by cosydney** [comment]
Alright, so I've manage to get it working as a single app, but now I'm trying to have it working with basePath. 

1) was already implemented in my version of the blog. 

and 2)

I tried different combinaison but still can't get it to work. Our blog section is served as 

axolo.co/blog

So our basePath is /blog for the app containing the blog. 

Do you know how I should set up the rewrites for it to work?


Here is how my next.config.js file looks like: 

```javascript
const withBundleAnalyzer = require('@next/bundle-analyzer')({
  enabled: process.env.ANALYZE === 'true',
})

module.exports = withBundleAnalyzer({
  basePath: '/blog',
  //.... some webpack stuff
  async rewrites() {
    return {
      beforeFiles: [
        // These rewrites are checked after headers/redirects
        // and before pages/public files which allows overriding
        // page files
        // we do the rewriting from blog to blog/index
        {
          source: '/',
          destination: '/index',
        },
      ],
    }
  },
})


```

**Comment by timlrx** [comment]
I didn't realize that the homepage was set to https://blog.axolo.co/blog instead of the more typical https://blog.axolo.co. For some reason there's a https://axolo.co/blog/index.xml created which should not be the case if you have already done (1). There's no need to do (2) if (1) is already done.

Can you check that this is similar on your side: https://github.com/timlrx/timlrx.com/blob/master/pages/blog/%5B...slug%5D.js#L36

Basically, you need to find where is the index.xml coming from and remove / fix it.

**Comment by timlrx** [comment]
Yup, looking at the xml file in more detail, it's actually the RSS feed. So one of the `generateRss` functions is saving as `/blog/index.xml`. You should change that to `/blog/feed.xml` or something else.

**Comment by cosydney** [comment]
@timlrx Thank you that solved the issue ! You rock
